source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
             HAL_ETH_DescAssignMemory( &( xEthHandle ), uxIndex, pucBuffer, NULL );
         }
 
+        #if ( ipconfigUSE_MDNS == 1 )
```

Comment:
Should we add && ( ipconfigUSE_IPv6 != 1 )

Reply:
No they are different MAC addresses

Reply:
Yes that is why Ipv4 check is needed.